src/hotspot/share/opto/vectorIntrinsics.cpp

```diff
@@ -1799,48 +1799,54 @@ bool LibraryCallKit::inline_vector_rearrange() {
     }
     return false; // not supported
   }
-  if (!arch_supports_vector(Op_VectorRearrange, num_elem, elem_bt, VecMaskNotUsed)) {
+
+  bool is_masked_op = argument(7)->bottom_type() != TypePtr::NULL_PTR;
+  bool use_predicate = is_masked_op;
+  if (is_masked_op &&
+      (mask_klass == NULL ||
+       mask_klass->const_oop() == NULL ||
+       !is_klass_initialized(mask_klass))) {
     if (C->print_intrinsics()) {
-      tty->print_cr("  ** not supported: arity=2 op=shuffle/rearrange vlen=%d etype=%s ismask=no",
-                    num_elem, type2name(elem_bt));
+      tty->print_cr("  ** mask_klass argument not initialized");
+    }
+  }
+  if (!arch_supports_vector(Op_VectorRearrange, num_elem, elem_bt, is_masked_op ? VecMaskUsePred : VecMaskNotUsed)) {
```

Comment:
For masked op, it's better to use the mask type `VecMaskUsePred | VecMaskUseLoad` here?

Reply:
If target does not support predicated rearrange instruction then a blend is the consume of mask, this looks correct.